src/rules/__tests__/require-types.test.ts

```diff
@@ -0,0 +1,113 @@
+import { RuleTester } from 'eslint';
+
+import php from '../../index';
+import { requireTypes } from '../require-types';
+
+const ruleTester = new RuleTester({
+	plugins: { php },
+	language: 'php/php',
+});
+
+ruleTester.run('require-types', requireTypes, {
+	valid: [
+		'<?php class Foo { public int $a = 1; }',
+		'<?php class Foo { public function bar(): int {} }',
+		'<?php class Foo { public function __construct() {} }',
+		'<?php class Foo { public function bar( int $param ): int {} }',
+		'<?php function foo(): int {}',
+		'<?php function foo( int $param ): int {}',
+		'<?php array_map(function(): int {}, []);',
+		'<?php array_map(function( int $param ): int {}, []);',
+	],
+	invalid: [
+		{
+			code: '<?php class Foo { public $bar; }',
+			errors: [
+				{
+					messageId: 'requireTypesForProperty',
+					data: { name: 'bar' },
+					line: 1,
+					column: 26,
+					endLine: 1,
+					endColumn: 30,
+				},
+			],
+		},
+		{
+			code: '<?php class Foo { public function bar() {} }',
+			errors: [
+				{
+					messageId: 'requireTypesForMethodReturnType',
+					data: { name: 'bar' },
+					line: 1,
+					column: 19,
+					endLine: 1,
+					endColumn: 43,
+				},
```

Comment:
maybe put the error only on the function name, because in this case the whole function will be red which is overwhelming (think about large functions)

also in other places, try to minimize the redness

Reply:
Not sure how to do so on closure without a name.

Reply:
hmm
put it on the `function` keyword?